resources/js/components/RegisterComponent.vue

```diff
@@ -0,0 +1,293 @@
+                                        name="password_confirmation"
+                                        required
+                                        autocomplete="new-password"
+                                        v-model="password_confirmation.val"
+                                    />
+                                    <span
+                                        class="invalid-feedback"
+                                        :class="{
+                                            block: password_confirmation.isValid
+                                        }"
+                                        role="alert"
+                                    >
+                                        <strong>{{
+                                            password_confirmation.msg
+                                        }}</strong>
+                                    </span>
+                                </div>
+                            </div>
+
+                            <div class="form-group row mb-0">
+                                <label
+                                    for="password-confirm"
+                                    class="col-md-4 col-form-label text-md-right"
+                                ></label>
+                                <div class="col-md-6 offset-md-4">
+                                    <button class="btn btn-primary pull-left">
+                                        Register
+                                    </button>
+                                </div>
+                            </div>
+                        </form>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+<script>
+export default {
+    data() {
+        return {
+            name: {
+                val: "",
+                msg: "",
+                isValid: false
+            },
+            email: {
+                val: "",
+                msg: "",
+                isValid: false
+            },
+            password: {
+                val: "",
+                isValid: false,
+                msg: ""
+            },
+            password_confirmation: {
+                val: "",
+                msg: "",
+                isValid: false
+            }
+        };
+    },
+    methods: {
+        register() {
+            let that = this;
+            this.axios
+                .post("/register", {
+                    name: this.name.val,
+                    email: this.email.val,
+                    password: this.password.val,
+                    password_confirmation: this.password_confirmation.val
+                })
+                .then(function(response) {
+                    // alert(response);
+                    window.location.href = "/task";
+                })
+                .catch(function(error) {
+                    if (error.response.data.errors.name !== undefined) {
```

Comment:
All this is not needed. Laravel already returns which fields failed along with the error messages for those fields. All this can just be a single line code i.e. `this.errors = error.response.data.errors`. This way you are populating `errors` property of vue with the entire errors array returned in the response. Then in the HTML above you can add `v-if` around the `errors` vue property. Example: `v-if="errors.name[0]"`